DataFormats/METReco/interface/HaloClusterCandidateEE.h

```diff
@@ -0,0 +1,84 @@
+#ifndef DATAFORMATS_METRECO_HALOCLUSTERCANDIDATEEE_H
+#define DATAFORMATS_METRECO_HALOCLUSTERCANDIDATEEE_H
+
+
+
+#include "DataFormats/DetId/interface/DetId.h"
+#include "DataFormats/EcalDetId/interface/EcalSubdetector.h"
+#include "DataFormats/EcalDetId/interface/EBDetId.h"
+#include "DataFormats/EcalDetId/interface/EEDetId.h"
+#include "DataFormats/GeometrySurface/interface/Cylinder.h"
+#include "DataFormats/GeometrySurface/interface/Plane.h"
+#include "DataFormats/GeometrySurface/interface/Cone.h"
+#include "DataFormats/GeometryVector/interface/GlobalPoint.h"
+#include "DataFormats/GeometryVector/interface/GlobalVector.h"
+#include "DataFormats/GeometryVector/interface/LocalPoint.h"
+#include "DataFormats/GeometryVector/interface/LocalVector.h"
+#include "DataFormats/HcalRecHit/interface/HcalRecHitCollections.h"
+#include "DataFormats/EcalRecHit/interface/EcalRecHitCollections.h"
+#include "DataFormats/Math/interface/deltaPhi.h"
+#include "Geometry/CaloGeometry/interface/CaloGeometry.h"
+#include "Geometry/Records/interface/CaloGeometryRecord.h"
+#include "FWCore/Framework/interface/Event.h"
+
+
+namespace reco {
+class HaloClusterCandidateEE {
```

Comment:
It would be better to reduce copy-pasted code between HaloClusterCandidateEE and HaloClusterCandidateEB.
Why not just simply have one HaloClusterCandidateECAL and fill EE and EB-specific values with some defaults and/or specify that it's EB or EE with a flag.
Given that vectors for EB and EE in EcalHaloData.h are in separate vectors, there should be no confusion which values can be checked.